appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m

```diff
@@ -74,12 +101,78 @@ + (void)register {
                   readOptions:(NSDictionary *)readOptionsMap
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject) {
-    MSReadOptions *readOptions = [self getReadOptions:readOptionsMap];
+    MSReadOptions *readOptions = [AppCenterReactNativeData getReadOptions:readOptionsMap];
     [MSData readDocumentWithID:documentID
                   documentType:[MSDictionaryDocument class]
                      partition:partition
                    readOptions:readOptions
-             completionHandler:[self dataCompletionHandler:kMSReadFailedErrorCode resolve:resolve reject:reject]];
+             completionHandler:[AppCenterReactNativeData dataCompletionHandler:kMSReadFailedErrorCode resolver:resolve rejecter:reject]];
+}
+
+RCT_EXPORT_METHOD(list:(NSString *)partition
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    [MSData listDocumentsWithType:[MSDictionaryDocument class] partition:partition completionHandler:^(MSPaginatedDocuments* _Nonnull documentWrappers) {
+        NSString *paginatedDocumentsId = [[NSUUID UUID] UUIDString];
+        _paginatedDocuments[paginatedDocumentsId] = documentWrappers;
+        NSMutableDictionary *paginatedDocumentsDict = [[NSMutableDictionary alloc] init];
+        NSMutableDictionary *currentPageDict = [[NSMutableDictionary alloc] init];
+        NSMutableArray *itemsArray = [[NSMutableArray alloc] init];
+        MSPage *currentPage = documentWrappers.currentPage;
+        if(currentPage.error) {
+            reject(kMSListFailedErrorCode, currentPage.error.description, currentPage.error);
+            return;
+        }
+        NSArray<MSDocumentWrapper *> *documents = currentPage.items;
+        for(MSDocumentWrapper *document in documents) {
```

Comment:
It might be more readable if we factor out the whole for-loop (`for(MSDocumentWrapper *document in documents)`) into a separate method. If you do that, please also move `NSMutableArray *itemsArray = [[NSMutableArray alloc] init];` inside the new method.

Reply:
Done, added [addDocumentsToArray](https://github.com/microsoft/appcenter-sdk-react-native/pull/639/files#diff-0348c1f142790e13c15d977bc9ab4a1fR240)